clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpCodec.java

```diff
@@ -15,74 +15,63 @@
  */
 package org.ehcache.clustered.common.messages;
 
+import org.ehcache.clustered.common.messages.ServerStoreOpMessage.AppendMessage;
+import org.ehcache.clustered.common.messages.ServerStoreOpMessage.GetAndAppendMessage;
+import org.ehcache.clustered.common.messages.ServerStoreOpMessage.GetMessage;
+import org.ehcache.clustered.common.messages.ServerStoreOpMessage.ReplaceAtHeadMessage;
+import org.ehcache.clustered.common.messages.ServerStoreOpMessage.ServerStoreOp;
 
 import java.nio.ByteBuffer;
-import java.nio.charset.Charset;
-
-import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.AppendMessage;
-import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.GetAndAppendMessage;
-import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.ReplaceAtHeadMessage;
 
 /**
  *
  */
-public class ServerStoreOpCodec {
+class ServerStoreOpCodec {
+
 
-  private static final Charset UTF_8 = Charset.forName("UTF-8");
+  private static final byte STORE_OP_CODE_SIZE = 1;
+  private static final byte CACHE_ID_LEN_SIZE = 4;
+  private static final byte KEY_SIZE = 8;
+  private static final byte CHAIN_LEN_SIZE = 4;
 
-  private static final byte MSG_TYPE_OFFSET = 1;
-  private static final byte STORE_OP_CODE_OFFSET = 1;
-  private static final byte CACHE_ID_LEN_OFFSET = 4;
-  private static final byte KEY_OFFSET = 8;
-  private static final byte CHAIN_LEN_OFFSET = 4;
+  private final ChainCodec chainCodec;
 
-  public static byte[] encode(ServerStoreOpMessage message) {
+  ServerStoreOpCodec() {
+    this.chainCodec = new ChainCodec();
+  }
+
+  public byte[] encode(ServerStoreOpMessage message) {
```

Comment:
The whole method would read better if each case was simply a call to a method doing the necessary work per message type.